wrappers/golang/pqc/ml-kem/config.go

```diff
@@ -0,0 +1,34 @@
+package mlkem
+
+import (
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/runtime"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/runtime/config_extension"
+)
+
+type MlKemConfig struct {
+	StreamHandle          runtime.Stream
+	IsAsync               bool
+	MessagesOnDevice      bool
+	EntropyOnDevice       bool
+	PublicKeysOnDevice    bool
+	SecretKeysOnDevice    bool
+	CiphertextsOnDevice   bool
+	SharedSecretsOnDevice bool
```

Comment:
These should all be private as the user doesn't need to update them since we update them internally in each functions Check
```suggestion
	messagesOnDevice      bool
	entropyOnDevice       bool
	publicKeysOnDevice    bool
	secretKeysOnDevice    bool
	ciphertextsOnDevice   bool
	sharedSecretsOnDevice bool
```